implants/lib/eldritch/src/sys/write_reg_impl.rs

```diff
@@ -0,0 +1,650 @@
+            
+        };
+        
+        let hive = RegKey::predef(ihive);
+        let (nkey, _ndisp) = hive.create_subkey(regpath)?;
+
+        match regtype.as_ref() {
+            "REG_NONE" => {
+                nkey.set_value(regname, &regvalue)?;
+            },
+            "REG_SZ" => nkey.set_value(regname, &regvalue)?,
+            "REG_EXPAND_SZ" => nkey.set_value(regname, &regvalue)?,
+            "REG_BINARY" => {
+                let data = RegValue{ vtype: REG_BINARY, bytes: regvalue.as_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_DWORD" => {
+                let parsed_value: u32 = u32::from_str_radix(&regvalue, 10)?;
+                let data = RegValue{ vtype: REG_DWORD, bytes: parsed_value.to_le_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_DWORD_BIG_ENDIAN" => {
+                let parsed_value: u32 = u32::from_str_radix(&regvalue, 10)?;
+                let data = RegValue{ vtype: REG_DWORD_BIG_ENDIAN, bytes: parsed_value.to_be_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_LINK" => {
+                nkey.set_value(regname, &regvalue)?;
+            },
+            "REG_MULTI_SZ" => {
+                let parsed_value: Vec<&str> = regvalue.split(',').collect();
+                nkey.set_value(regname, &parsed_value)?;
+            },
+            "REG_RESOURCE_LIST" => {
+                let data = RegValue{ vtype: REG_RESOURCE_LIST, bytes: regvalue.as_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_FULL_RESOURCE_DESCRIPTOR" => {
+                let data = RegValue{ vtype: REG_FULL_RESOURCE_DESCRIPTOR, bytes: regvalue.as_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_RESOURCE_REQUIREMENTS_LIST" => {
+                let data = RegValue{ vtype: REG_RESOURCE_REQUIREMENTS_LIST, bytes: regvalue.as_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            "REG_QWORD" => {
+                let parsed_value: u64 = u64::from_str_radix(&regvalue, 10)?;
+                let data = RegValue{ vtype: REG_QWORD, bytes: parsed_value.to_le_bytes().to_vec()};
+                nkey.set_raw_value(regname, &data)?;
+            },
+            _ => return Err(anyhow::anyhow!("RegType can only be one of the following values - REG_NONE, REG_SZ, REG_EXPAND_SZ, REG_BINARY, REG_DWORD, REG_DWORD_BIG_ENDIAN, REG_LINK, REG_MULTI_SZ, REG_RESOURCE_LIST, REG_RESOURCE_LIST, REG_FULL_RESOURCE_DESCRIPTOR, REG_QWORD. ")),
+        };
+
+        Ok(true)
+    }
+}
+
+pub fn write_reg_int(reghive: String, regpath: String, regname: String, regtype:String, regvalue: u32 ) -> Result<bool>  {
```

Comment:
I would split each of these into a separate impl file.
`write_reg_int_impl.rs`, `write_reg_str_impl.rs`, `write_reg_hex_impl.rs`

Reply:
thought about doing that. but assumed it'll be cleaner in one page. can fix quickly no problem